eZ/Publish/Core/MVC/Symfony/Matcher/ContentBased/Identifier/ContentType.php

```diff
@@ -52,14 +53,18 @@ public function matchContentInfo(ContentInfo $contentInfo)
 
     public function match(View $view)
     {
-        if (!$view instanceof ContentValueView) {
+        if ($view instanceof ContentTypeView) {
+            $contentTypeIdentifier = $this->repository
+                ->getContentTypeService()
+                ->loadContentType($view->getContentTypeId())->identifier;
```

Comment:
> Unless it is possible to compile the mapping into container cache and reuse that in mvc layer where this is needed in relation to config, as config changes will trigger container cache refresh anyway.

If you go into the compiler layer, then I expect that _a lot_ is possible :-)

But we can, and should, refactor/optimize the matchers. This is just a glimpse of what we should do, but we can do better.